src/pages/admin/categories/AddEditCategory/index.js

```diff
@@ -1,18 +1,64 @@
-import React, { useState } from 'react';
-import { useLocation } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+import { useLocation, useParams, useHistory } from 'react-router-dom';
 import { Card, Button, Form } from 'react-bootstrap';
+import { useForm, Controller } from 'react-hook-form';
 import { BsFillArrowLeftSquareFill } from 'react-icons/bs';
 import { CgMenuCake } from 'react-icons/cg';
+import { useToast } from '../../../../hooks/useToast';
+import Spinner from 'react-bootstrap/Spinner';
 
 import style from './index.module.scss';
 
 import ChangeLocation from '../../../../components/ChangeLocation';
+import CategoryApi from '../../../../api/Category';
 
 const AddEditCategory = () => {
   const location = useLocation();
   const [show, setShow] = useState(false);
   const handleClose = () => setShow(false);
   const handleShow = () => setShow(true);
+  const { control, handleSubmit } = useForm();
+  const [errors, setErrors] = useState({});
+  const [submitStatus, setSubmitStatus] = useState(false);
+  const { category_id } = useParams();
+  const [category, setCategory] = useState(null);
+
+  const history = useHistory();
+  const toast = useToast();
+
+  useEffect(() => {
+    if (location.pathname !== '/admin/add-category') {
+      CategoryApi.show({ categoryId: category_id }).then(({data}) => {
+        setCategory(data.data);
+      });
+    }
+  },[]);
+
+
+  const handleOnSubmit = async ({ name, description }) => {
+    setSubmitStatus(true);
+    setErrors('');
+
+    try {
+      if (location.pathname === '/admin/add-category') {
+        toast('Processing', 'Adding a Category...');
+        CategoryApi.store( name, description ).then(() => {
+          toast('Success', 'Successfully Added Category.');
+          history.push('/admin/categories');
+        });
+      } else { 
+        toast('Processing', 'Updating Category...');
+        CategoryApi.update(name, description, category_id).then(() => {
+          toast('Success', 'Successfully Updated Category.');
+          history.push('/admin/categories');
+        });
+      }
```

Comment:
use else if for the other pathname.

the else should show an error saying something like "invalid url"